app/components/TrackCard/index.js

```diff
@@ -11,35 +11,71 @@ import { Card } from '@mui/material';
 import T from '@components/T';
 import If from '@components/If';
 import isEmpty from 'lodash/isEmpty';
+import { useLocation } from 'react-router-dom';
 
 const CustomCard = styled(Card)`
   && {
     margin: 1rem 0;
     padding: 1rem;
+    min-height: 270px;
   }
 `;
 
-export function TrackCard({ collectionName, artistName, collectionPrice, artworkUrl100 }) {
-  return (
-    <CustomCard data-testid="repo-card">
-      
-      <If condition={!isEmpty(collectionName)} otherwise={<T data-testid="name-unavailable" id="repo_name_unavailable" />}>
-        <div ><img src={artworkUrl100} alt={collectionName} />
-        <T data-testid="name" id="track_name" values={{ trackName: collectionName }} />
-        </div>
-       
-      </If>
-      <If
-        condition={!isEmpty(artistName)}
-        otherwise={<T data-testid="fullName-unavailable" id="repo_full_name_unavailable" />}
-      >
-        <T data-testid="fullName" id="artist_name" values={{ artist_name: artistName }} />
-      </If>
-      <If condition={collectionPrice} otherwise={<T data-testid="stargazers-unavaiable" id="repo_stars_unavailable" />}>
-        <T data-testid="stargazers" id="total_cost" values={{ price: collectionPrice }} />
-      </If>
-    </CustomCard>
-  );
+export function TrackCard({ collectionName, artistName, collectionPrice, artworkUrl100, description }) {
+  const location = useLocation();
+  console.log(location.state);
+  if(isEmpty(collectionName) && !isEmpty(location.state)){
```

Comment:
Avoid defining two different types of ui in the same component function itself. Make one of the condition component code into a different component. As much as possible, try to follow DRY principle that is **Don’t Repeat Yourself.**